Yoast/javascript
# Allow custom SvgIcon components to be created

## Summary

Iteration on https://github.com/Yoast/javascript/pull/261.

Allows you to create `SvgIcon` component with a custom set of icons.

```
/* YourIconComponent.js */
import { createSvgIconComponent } from "@yoast/components";

const icons = { /* Your icons */ };

export default createSvgIconComponent( icons );
```

```
/* YourExtendedIconComponent.js */
import { createSvgIconComponent, icons } from "@yoast/components";

const icons = Object.assign( { /* Your icons */ }, icons );

export default createSvgIconComponent( icons );
```

## Relevant technical choices:

* This PR will make sure the current API of `SvgIcon` stay unchanged, because you will no longer be able to provide an `iconSet` props. **Therefore, it should be merged into 11.3.**

## Test instructions

This PR can be tested by following these steps:

* See https://github.com/Yoast/javascript/pull/261.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * https://github.com/Yoast/javascript/pull/261

## UI changes

## Quality assurance


Fixes #
